notebooks/Notebook3.md

```diff
@@ -2,33 +2,56 @@
+
+After some sleep I figured that there must be another online sheet like resource that uses simple http requests. And after some research light got shed on spreadsheet.com and their open API. Now the easy part of using spreadsheet.com and their api Is that I can send http requests with no hassle. The challenge of using spreadsheet.com is that they require their input of data to be in JSON. Now if you have no background in JSON this gets very confusing. Luckily I do have some background. The challenges that now lay ahead of me are converting my pre-existing data structure into JSON. I spent the rest of the week experimenting with different strategies to do so. I will provide a file that has some of my test code for this. It is very rough and unorganized. But I believe I have found and made some methods that will be able to get the job done. Next week I will plan on finalizing this aspect and making the code and the inputs into the sheet look nice. ([example code not yet in project](https://github.com/hmc-cs111-spring2023/Bbecker8-Artifact/blob/main/.devcontainer/Examplecode))
+
 
 ## Questions
 
 **What is the most pressing issue for your project? What design decision do
 you need to make, what implementation issue are you trying to solve, or how
 are you evaluating your design and implementation?**
 
+The most pressing issue for me now is working to turn my pre-existing data structure into JSON.
+
+
 **What questions do you have for your critique partners? How can they best help
 you?**
 
+I don’t have any question at this time
+
 **How much time did you spend on the project this week? If you're working in a
 team, how did you share the work?**
 
+I spent roughly 10 hours this week
+
 **Compared to what you wrote in your contract about what you want to get out of this
 project, how did this week go?**
+
+At this point I think I am still a week ahead of where I said I was going to be but I have some extra thought on extra things that I want to achieve in this project. 
```

Comment:
Great job maintaining your pace since the first week, I like how well your DSL has come together including the new interface with error handling progress this week, it made it seem much more concrete to see what users will be interacting with and how they will be able to work through any issues with their inputs.